gpbft/emulator/chain.go

```diff
@@ -0,0 +1,51 @@
+package emulator
```

Comment:
So I suggest moving this out of the gpbft package and as a peer of `sim`. If we were using Rust crates or some other decent dependency/package system, one would want to be able to depend on the protocol code without necessarily pulling this in.